panther_hardware_interfaces/include/panther_hardware_interfaces/roboteq_error_filter.hpp

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_HARDWARE_INTERFACES_ROBOTEQ_ERROR_FILTER_HPP_
+#define PANTHER_HARDWARE_INTERFACES_ROBOTEQ_ERROR_FILTER_HPP_
+
+#include <atomic>
+#include <vector>
+
+namespace panther_hardware_interfaces
+{
+
+class ErrorFilter
+{
+public:
+  ErrorFilter(const unsigned max_error_count) : max_error_count_(max_error_count) {}
+
+  bool IsError() const { return error_; };
+
+  /**
+   * @brief Updates the error count, if the number of consecutive errors exceeds the max
+   * threshold error is set
+   */
+  void UpdateError(const bool current_error)
+  {
+    if (current_error) {
+      ++current_error_count_;
+      if (current_error_count_ >= max_error_count_) {
+        error_ = true;
+      }
+    } else {
+      current_error_count_ = 0;
+    }
+  }
+
+  void ClearError()
+  {
+    error_ = false;
+    current_error_count_ = 0;
+  }
+
+private:
+  const unsigned max_error_count_;
+  unsigned current_error_count_ = 0;
+  bool error_ = false;
+};
+
+enum class ErrorsFilterIds { READ_SDO, WRITE_SDO, READ_PDO, ROBOTEQ_DRIVER };
```

Comment:
```suggestion
enum class ErrorsFilterIds {
  READ_SDO = 0,
  WRITE_SDO,
  READ_PDO,
  ROBOTEQ_DRIVER,
};
```
again comma does the trick